workflow/snakemake_rules/nextstrain_auspice_config.smk

```diff
@@ -0,0 +1,160 @@
+rule make_auspice_config:
+    """
+    This rule is only intended to be run with `nextstrain-open` or `nextstrain-gisaid`
+    profiles!
+    """
+    message: "Making a custom auspice config."
+    output:
+        "results/{build_name}/auspice_config.json"
+    benchmark:
+        "benchmarks/make_auspice_config_{build_name}.txt"
+    run:
+        import json
+        is_gisaid = "gisaid" in config["inputs"]
+        build_name = wildcards.build_name
+
+        ## What are the parameters which vary across builds?
+        default_geo_resolution = "country" if build_name in ["reference", "global", "africa", "south-america", "asia", "europe"] else "division"
+        default_map_triplicate = True if build_name in ["reference", "global"] else False
+        data_provenance = [{"name": "GISIAD"}] if is_gisaid else [{"name": "GenBank", "url": "https://www.ncbi.nlm.nih.gov/genbank/"}]
+        gisaid_clade_coloring = {"key": "GISAID_clade", "title": "GISAID Clade", "type": "categorical"} if is_gisaid else None
+        gisaid_epi_isl_coloring = {"key": "gisaid_epi_isl", "type": "categorical"} if is_gisaid else None
+        location_coloring = {"key": "location", "title": "Location", "type": "categorical"} if is_gisaid else None
+        location_filter = "location" if is_gisaid else None
+        originating_lab_filter = "originating_lab" if is_gisaid else None
+        submitting_lab_filter  = "submitting_lab" if is_gisaid else None
```

Comment:
These conditionals assume GenBank/Open if not GISAID instead of explicitly testing for GenBank/Open, which is the better/more future-proof practice. Would prefer to explicitly test, plus also make the test against `inputs` consider the whole value and not just single-key presence. There's a number of ways to make this approach syntactically palatable, but here's one example:

```suggestion
        input_set = set(config["inputs"])
        build_name = wildcards.build_name

        ## What are the parameters which vary across builds?
        default_geo_resolution = "country" if build_name in ["reference", "global", "africa", "south-america", "asia", "europe"] else "division"
        default_map_triplicate = True if build_name in ["reference", "global"] else False

        if input_set == {"gisaid"}:
            data_provenance = [{"name": "GISIAD"}]
            gisaid_clade_coloring = {"key": "GISAID_clade", "title": "GISAID Clade", "type": "categorical"}
            gisaid_epi_isl_coloring = {"key": "gisaid_epi_isl", "type": "categorical"}
            location_coloring = {"key": "location", "title": "Location", "type": "categorical"}
            location_filter = "location"
            originating_lab_filter = "originating_lab"
            submitting_lab_filter  = "submitting_lab"
        elif input_set == {"open"}:
            data_provenance = [{"name": "GenBank", "url": "https://www.ncbi.nlm.nih.gov/genbank/"}]
            gisaid_clade_coloring = None
            gisaid_epi_isl_coloring = None
            location_coloring = None
            location_filter = None
            originating_lab_filter = None
            submitting_lab_filter  = None
        else:
            raise Exception(f"rule auspice_config doesn't know how to handle inputs: {input_set}")
```

Reply:
Sure!